src/content/1.7/modules/sample_modules/view-order-hooks-demo.md

```diff
@@ -0,0 +1,522 @@
+    exit;
+}
+
+class DemoViewOrderHooks extends Module
+{
+    public function __construct()
+    {
+        $this->name = 'demovieworderhooks';
+        $this->author = 'PrestaShop';
+        $this->version = '1.0.0';
+        $this->ps_versions_compliancy = ['min' => '1.7.7.0', 'max' => _PS_VERSION_];
+
+        parent::__construct();
+
+        $this->displayName = $this->l('Demo view order hooks');
+        $this->description = $this->l('Demonstration of new hooks in PrestaShop 1.7.7 order view page');
+    }
+
+    public function install()
+    {
+        if (!parent::install()) {
+            return false;
+        }
+
+        $installer = InstallerFactory::create();
+
+        return $installer->install($this);
+    }
+
+    public function uninstall()
+    {
+        $installer = InstallerFactory::create();
+
+        return $installer->uninstall() && parent::uninstall();
+    }
+
+}
+```
+
+Let's create `Signature` entity class:
+
+```php
+<?php
+
+declare(strict_types=1);
+
+namespace PrestaShop\Module\DemoViewOrderHooks\Entity;
+
+use Doctrine\ORM\Mapping as ORM;
+
+/**
+ * @ORM\Table()
+ * @ORM\Entity()
+ */
+class Signature
+{
+    /**
+     * @var int|null
+     *
+     * @ORM\Id
+     * @ORM\Column(name="id_signature", type="integer")
+     * @ORM\GeneratedValue(strategy="AUTO")
+     */
+    private $id;
+
+    /**
+     * @var int
+     *
+     * @ORM\Column(name="id_order", type="integer")
+     */
+    private $orderId;
+
+    /**
+     * @var string
+     *
+     * @ORM\Column(name="filename", type="string")
+     */
+    private $filename;
+
+    /**
+     * @return int|null
+     */
+    public function getId(): ?int
+    {
+        return $this->id;
+    }
+
+    /**
+     * @param int $id
+     *
+     * @return Signature
+     */
+    public function setId(int $id): Signature
+    {
+        $this->id = $id;
+
+        return $this;
+    }
+
+    /**
+     * @return string
+     */
+    public function getFilename(): string
+    {
+        return $this->filename;
+    }
+
+    /**
+     * @param string $filename
+     *
+     * @return Signature
+     */
+    public function setFilename(string $filename): Signature
+    {
+        $this->filename = $filename;
+
+        return $this;
+    }
+
+    /**
+     * @return int
+     */
+    public function getOrderId(): int
+    {
+        return $this->orderId;
+    }
+
+    /**
+     * @param int $orderId
+     *
+     * @return Signature
+     */
+    public function setOrderId(int $orderId): Signature
+    {
+        $this->orderId = $orderId;
+
+        return $this;
+    }
+}
+
+```
+
+Lets create `SignatureRepository` class:
+
+```php
+<?php
+declare(strict_types=1);
+
+namespace PrestaShop\Module\DemoViewOrderHooks\Repository;
+
+use Doctrine\ORM\EntityRepository;
+
+class SignatureRepository extends EntityRepository
+{
+}
+```
+
+Lets create services configuration for the above classes to use dependency injection (DI) in `config/services.yml`:
```

Comment:
Mentioning '(DI)' sounds useless if you are not using it later as abbreviation. Maybe link to DI docs would be usefull